### PR TITLE
swarm: plan bee for content storage and distribution on web3

### PR DESCRIPTION
**Pull request by fjl**
This change imports the Swarm protocol codebase. Compared to the 'swarm'
branch, a few mostly cosmetic changes had to be made:

* The various redundant log message prefixes are gone. Fixes https://github.com/ethereum/go-ethereum/issues/2345
* All files now have LGPLv3 license headers.
* Minor code changes were needed to please go vet and make the tests
  pass on Windows.
* Further changes were required to adapt to the go-ethereum develop
  branch and its new Go APIs.

Some code has not (yet) been brought over:

* swarm/cmd/bzzhash: will reappear as cmd/bzzhash later
* swarm/cmd/bzzup.sh: will be reimplemented in cmd/bzzup
* swarm/cmd/makegenesis: will reappear somehow
* swarm/examples/album: the frontend will be submitted later
* swarm/examples/filemanager: ditto
* swarm/examples/files: will not be merged
* swarm/test/*: will not be merged
* swarm/services/swear: will reappear as contracts/swear when needed

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Mon Aug 29 20:05:55 UTC 2016_

File: contracts/chequebook/cheque.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/logger"
 	"github.com/ethereum/go-ethereum/logger/glog"
+	"github.com/ethereum/go-ethereum/swarm/services/swap/swap"
```
Comment (zelig) [inline contracts/chequebook/cheque.go]: why are we reverting this change. was it not better?

**Comment by fjl** [comment]
A few suggestions that I'd like to discuss:

* Should the swarm package tree be called "bzz" instead?
* "swarm/services/swap" -> "bzz/swap"? swap is the only thing left in services/

**Comment by zelig** [comment]
It used to be bzz and because of whisper was changed to swarm :)
I think  people will find the  swarm name easier to understand

yes I am less convinced about services subdir now myself, but there will be other services (sync,  depo, scan and repair, swindle). Since these are  protocol submodules we could put them under network. But top-level does not align  with the rest of the directory structure

**Comment by zelig** [comment]
:+1:

**Pull request merged**